managed/src/main/java/com/yugabyte/yw/commissioner/tasks/UniverseTaskBase.java

```diff
@@ -314,11 +315,19 @@ public SubTaskGroup createDestroyServerTasks(Collection<NodeDetails> nodes,
       // Check if the private ip for the node is set. If not, that means we don't have
       // a clean state to delete the node. Log it and skip the node.
       if (node.cloudInfo.private_ip == null) {
-        LOG.warn(String.format("Node %s doesn't have a private IP. Skipping node delete.",
-                               node.nodeName));
+        LOG.warn(String.format("Node %s doesn't have a private IP. Skipping node delete.", node.nodeName));
         // Free up the node so that the client can use the instance to create another universe.
-        NodeInstance providerNode = NodeInstance.getByName(node.nodeName);
-        providerNode.clearNodeDetails();
+        Universe universe = Universe.get(taskParams().universeUUID);
```

Comment:
You don't need the universe. The node details class itself has a field `cloudInfo.cloud` that you can use for this check.